tools-local/tasks/mobile.tasks/WasmAppBuilder/WasmAppBuilder.cs

```diff
@@ -76,62 +108,56 @@ public override bool Execute ()
 
-                var i = 0;
-                foreach (var item in FilesToIncludeInFileSystem)
-                {
-                    string? targetPath = item.GetMetadata("TargetPath");
-                    if (string.IsNullOrEmpty(targetPath))
-                    {
-                        targetPath = Path.GetFileName(item.ItemSpec);
-                    }
+        if (FilesToIncludeInFileSystem != null)
+        {
+            string supportFilesDir = Path.Join(AppDir, "supportFiles");
+            Directory.CreateDirectory(supportFilesDir);
 
-                    // We normalize paths from `\` to `/` as MSBuild items could use `\`.
-                    targetPath = targetPath.Replace('\\', '/');
+            var i = 0;
+            foreach (var item in FilesToIncludeInFileSystem)
+            {
+                string? targetPath = item.GetMetadata("TargetPath");
+                if (string.IsNullOrEmpty(targetPath))
+                {
+                    targetPath = Path.GetFileName(item.ItemSpec);
+                }
 
-                    var generatedFileName = $"{i++}_{Path.GetFileName(item.ItemSpec)}";
+                // We normalize paths from `\` to `/` as MSBuild items could use `\`.
+                targetPath = targetPath.Replace('\\', '/');
 
-                    File.Copy(item.ItemSpec, Path.Join(supportFilesDir, generatedFileName), true);
+                var generatedFileName = $"{i++}_{Path.GetFileName(item.ItemSpec)}";
 
-                    var actualItemName = "supportFiles/" + generatedFileName;
+                File.Copy(item.ItemSpec, Path.Join(supportFilesDir, generatedFileName), true);
 
-                    sw.WriteLine("\t\t{");
-                    sw.WriteLine("\t\t\tbehavior: \"vfs\",");
-                    sw.WriteLine($"\t\t\tname: \"{actualItemName}\",");
-                    sw.WriteLine($"\t\t\tvirtual_path: \"{targetPath}\",");
-                    sw.WriteLine("\t\t},");
-                }
+                var asset = new VfsEntry  {
+                    name = $"supportFiles/{generatedFileName}",
+                    virtual_path = targetPath
+                };
+                config.assets.Add(asset);
             }
+        }
 
-            var enableRemote = (RemoteSources != null) && (RemoteSources!.Length > 0);
-            var sEnableRemote = enableRemote ? "true" : "false";
+        var enableRemote = (RemoteSources != null) && (RemoteSources!.Length > 0);
 
-            sw.WriteLine($"\t\t{{ behavior: \"icu\", name: \"icudt.dat\", load_remote: {sEnableRemote} }},");
-            sw.WriteLine($"\t\t{{ behavior: \"vfs\", name: \"dotnet.timezones.blat\", virtual_path: \"/usr/share/zoneinfo/\" }}");
-            sw.WriteLine ("\t],");
+        config.assets.Add(new IcuData() { name = "icudt.dat", load_remote = enableRemote });
+        config.assets.Add(new VfsEntry() { name = "dotnet.timezones.blat", virtual_path = "/usr/share/zoneinfo/"});
```

Comment:
nit: the parens are not needed

```suggestion
        config.assets.Add(new IcuData { name = "icudt.dat", load_remote = enableRemote });
        config.assets.Add(new VfsEntry { name = "dotnet.timezones.blat", virtual_path = "/usr/share/zoneinfo/"});
```